definitions/pipelines/immuno.cwl

```diff
@@ -50,6 +50,10 @@ inputs:
         type: File
     ribosomal_intervals:
         type: File
+    kallisto_kmer_size:
+        type: int?
```

Comment:
This param is configurable for kallisto, but not exposed in the CWL (intentionally, because we've never seen any reason to change it!) This implementation is a little problematic because it would allow the user to set this value for pizzly, but that wouldn't propagate through to kallisto, which could cause a mismatch.  

I think we suffer from parameter overload already, and this is one example of where we can make an opinionated choice for the end user.  My suggestion would be to hardcode the default value of 31 at the level of the pizzly and kallisto tools, but not expose it upstream. Happy to discuss if you disagree!

Reply:
Isn't letting the user supply their own kallisto index in effect exposing this parameter to them already?

Reply:
Shit, yeah I guess it is. so I guess we really do need to expose this and the right way to fix it is to add it as an input to the kallisto cwl (and up the chain) preferably with a default value of 31

Reply:
Since the workflow/kallisto tool don't generate an index each time and instead accept a pre-existing one, does it make sense to add this upstream, since it won't do anything? I checked the kallisto source code and used it to make a small script that can pull the kmer size out of an index file. How do we feel about adding that to the workflow and dropping the kmer parameter entirely, since the index is always an input anyway?

Reply:
I like it!